server/queue/queue-rabbitmq/src/main/java/org/apache/james/queue/rabbitmq/Dequeuer.java

```diff
@@ -0,0 +1,182 @@
+        private final long deliveryTag;
+        private final Mail mail;
+
+        private RabbitMQMailQueueItem(RabbitClient rabbitClient, long deliveryTag, Mail mail) {
+            this.rabbitClient = rabbitClient;
+            this.deliveryTag = deliveryTag;
+            this.mail = mail;
+        }
+
+        @Override
+        public Mail getMail() {
+            return mail;
+        }
+
+        @Override
+        public void done(boolean success) throws MailQueue.MailQueueException {
+            try {
+                rabbitClient.ack(deliveryTag);
+            } catch (IOException e) {
+                throw new MailQueue.MailQueueException("Failed to ACK " + mail.getName() + " with delivery tag " + deliveryTag, e);
+            }
+        }
+    }
+
+    private static final int TEN_MS = 10;
+
+    private final MailQueueName name;
+    private final RabbitClient rabbitClient;
+    private final Store<MimeMessage, MimeMessagePartsId> mimeMessageStore;
+    private final BlobId.Factory blobIdFactory;
+    private final ObjectMapper objectMapper;
+    private final Metric dequeueMetric;
+
+    Dequeuer(MailQueueName name, RabbitClient rabbitClient, Store<MimeMessage, MimeMessagePartsId> mimeMessageStore, BlobId.Factory blobIdFactory, ObjectMapper objectMapper, MetricFactory metricFactory) {
+        this.name = name;
+        this.rabbitClient = rabbitClient;
+        this.mimeMessageStore = mimeMessageStore;
+        this.blobIdFactory = blobIdFactory;
+        this.objectMapper = objectMapper;
+        this.dequeueMetric = metricFactory.generate(DEQUEUED_METRIC_NAME_PREFIX + name.asString());
+    }
+
+    MailQueue.MailQueueItem deQueue() throws MailQueue.MailQueueException {
+        GetResponse getResponse = pollChannel();
+        MailDTO mailDTO = toDTO(getResponse);
+        Mail mail = toMail(mailDTO);
+        dequeueMetric.increment();
+        return new RabbitMQMailQueueItem(rabbitClient, getResponse.getEnvelope().getDeliveryTag(), mail);
+    }
+
+    private MailDTO toDTO(GetResponse getResponse) throws MailQueue.MailQueueException {
+        try {
+            return objectMapper.readValue(getResponse.getBody(), MailDTO.class);
+        } catch (IOException e) {
+            throw new MailQueue.MailQueueException("Failed to parse DTO", e);
+        }
+    }
+
+    private GetResponse pollChannel() {
+        return new AsyncRetryExecutor(Executors.newSingleThreadScheduledExecutor())
+            .withFixedRate()
+            .withMinDelay(TEN_MS)
+            .retryOn(NoMailYetException.class)
+            .getWithRetry(this::singleChannelRead)
+            .join();
+    }
+
+    private GetResponse singleChannelRead() throws IOException {
+        return rabbitClient.poll(name)
+            .filter(getResponse -> getResponse.getBody() != null)
+            .orElseThrow(NoMailYetException::new);
+    }
+
+    private Mail toMail(MailDTO dto) throws MailQueue.MailQueueException {
```

Comment:
it's not a `toMail` method as it read from the mimeMessageStore. It could be a `loadMail` maybe. And maybe MailDTO is not a very good type name because it's not really a Mail representation but rather a MailReference or alike.